android/src/app/organicmaps/widget/menu/NavMenu.java

```diff
@@ -208,11 +210,49 @@ private void updateSpeedView(@NonNull RoutingInfo info)
 
     Pair<String, String> speedAndUnits = StringUtils.nativeFormatSpeedAndUnits(last.getSpeed());
 
+    // (temporarily) show the speed limit together with the current speed.
+    String speedIndicator = formatSpeed(speedAndUnits, info);
+    mSpeedValue.setText(speedIndicator);
+
     mSpeedUnits.setText(speedAndUnits.second);
-    mSpeedValue.setText(speedAndUnits.first);
     mSpeedViewContainer.setActivated(info.isSpeedLimitExceeded());
   }
 
+  private String formatSpeed(Pair<String, String> speedAndUnits, RoutingInfo info)
+  {
+    double currentSpeedMps = Double.parseDouble(speedAndUnits.first);
+    String speedIndicator = String.format("%.0f", currentSpeedMps);
+
+    double speedLimit = convertMpsIntoUnits(info.speedLimit, speedAndUnits.second);
+    if (speedLimit > 0.0)
+    {
+      speedIndicator += String.format("/%.0f", speedLimit);
+    }
+
+    return speedIndicator;
+  }
+
+  private double convertMpsIntoUnits(double speedInMps, String units)
+  {
+    double result;
+
+    switch (units)
+    {
+    case "km/h":
+      // 1 m/s = 3600s/1000m = 18/5 km/h
+      result = speedInMps * 18 / 5;
```

Comment:
A bit off-topic but I'm wondering why do people often use 18/5 when converting to kph when it's just 3.6? Is it easier to remember or what?

Reply:
It is easier to copy/paste from a search result for "m/s to mph" ;)

Reply:
`3600 / 1000` is way better and more obvious than 18/5

Reply:
@alensiljak also if you change it in one place change in every other to keep it consistent.

Reply:
Seems that you guys have more time to work on this than I do. :D

Reply:
Fix the comment too.